common/browser_process_meta.go

```diff
@@ -0,0 +1,72 @@
+package common
+
+import (
+	"errors"
+	"os"
+
+	"github.com/grafana/xk6-browser/storage"
+)
+
+// ErrUnknownRemoteBrowserProcessPid indicates that the pid for the browser process is unknown
+// as the browser is being executed in a remote machine.
+var ErrUnknownRemoteBrowserProcessPid = errors.New("browser process is remote and pid unknown")
+
+// BrowserProcessMeta handles the metadata associated with
+// a browser process, especifically, the OS process handle
+// and the associated browser data directory.
+type BrowserProcessMeta interface {
+	Pid() (int, error)
```

Comment:
Returning an error from a getter field does not seem to be the right design here. Since this method is primarily needed by `*Browser.Close()` (?). The other usage where we check this error is `*Browser.launch()`, but it doesn't seem necessary since `launch` always returns a PID, and we won't call `launch` when using `Connect()`. It seems we're unnecessarily checking an error in the latter.

Also, only the local browser meta uses `Cleanup()`. Could we instead inject the clean-up as a function value to change its behavior? Even further, do we know if we need a cleanup method on the remote browser?

Reply:
Agree that returning an error from a getter might not be the most elegant solution, but the idea was to provide an abstraction that would be safe to use for future cases also. So, with the current implementation we could make the changes that you mention, but we take the risk of "misusing" the `Pid` method in the future, assuming that it will always return a valid pid when in some cases, when using connect, it will not.
In any case, I don't have strong opinion, if you think this is unnecessary and that we can "take that risk" and handle it ourselves in future implementations and tests, that's ok! I will change it.

Regarding cleanup, the argument is similar as the previous point, the idea was to provide a safe abstraction that would work minimizing the changes on the current implementation. When you say passing `cleanup` as function, you mean to the browser process constructors?

Reply:
I would take that risk since I don't think it's a big one, but I'd leave the decision to you 👍

Reply:
It's ok for me! Will change this to NOT return an error, unless @ankur22 disagrees.

Reply:
Yep, let's try without it.

Reply:
Changed. See: [510346a](https://github.com/grafana/xk6-browser/pull/800/commits/510346a0285e638acc116ea3e0d7e0638ffa0fc3) and [f98a35f](https://github.com/grafana/xk6-browser/pull/800/commits/f98a35f4c89c66066b37f2683a519a8383fc2c2b).

Reply:
@ka3de, thanks 🙇 

> When you say passing cleanup as function, you mean to the browser process constructors?

I mean injecting a cleanup function to the browser process instead of making this an interface.

We have two operating modes: remote and local; and I'm not sure it deserves an interface. Interface=slow and brings more complexity. Concrete=faster, direct, and easy to jump in the code editor. So to wrap up, another idea, `*BrowserProcess.Cleanup()` can decide whether to call the storage's `Cleanup` function without needing an interface.

Reply:
I explored this option initially (mentioned in the PR description), but I discarded it in favor of an interface because it provides a better abstraction and more safety measures, in this case for the `os.Process` and `dataDir`, which is important considering that we do not have any abstraction for `BrowserProcess`, so any other component can reach the internal `os.Process` property of browser process, which in some situations will be `nil`. So I'm in favor of current implementation for these reasons, but let's discuss.

Reply:
Alright, resolving if everyone is OK. @ankur22 @ka3de